server/i18n.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/nicksnyder/go-i18n/v2/i18n"
+	"github.com/pkg/errors"
+	"golang.org/x/text/language"
+)
+
+// initI18nBundle loads all localization files in i18n into a bundle and return this
+func (p *Plugin) initI18nBundle() (*i18n.Bundle, error) {
+	bundle := i18n.NewBundle(language.English)
+	bundle.RegisterUnmarshalFunc("json", json.Unmarshal)
+
+	bundlePath, err := p.API.GetBundlePath()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get bundle path")
+	}
+
+	i18nDir := filepath.Join(bundlePath, "assets", "i18n")
+	files, err := ioutil.ReadDir(i18nDir)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to open i18n directory")
+	}
+
+	for _, file := range files {
+		if file.Name() == "en.json" {
+			continue
+		}
+		_, err = bundle.LoadMessageFile(filepath.Join(i18nDir, file.Name()))
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to load message file %s", file.Name())
+		}
+	}
+
+	return bundle, nil
+}
+
+// getUserLocalizer returns a localizer that localizes in the users locale
+func (p *Plugin) getUserLocalizer(userID string) *i18n.Localizer {
+	user, err := p.API.GetUser(userID)
+	if err != nil {
+		p.API.LogWarn("Failed get user's locale", "error", err.Error())
+		return p.getServerLocalizer()
+	}
+
+	return i18n.NewLocalizer(p.i18nBundle, user.Locale)
+}
+
+// getServerLocalizer returns a localizer that localizes in the default server locale
+func (p *Plugin) getServerLocalizer() *i18n.Localizer {
+	return i18n.NewLocalizer(p.i18nBundle, *p.API.GetConfig().LocalizationSettings.DefaultServerLocale)
```

Comment:
I'm wondering if `DefaultClientLocale` is a better value to use there. Is the default locale for users that freshly log into the system and hence just by most users.

Reply:
I don't think so, for me, the expectation is, if you have configure the server in a language, all the messages generated by the server itself, is going to be in that language, unless you can properly figure out who you are sending the message. I don't think the default client locale is for decide how to translate server generated strings (If that is the case, what is the default server locale for?)

Reply:
According to the docs: `DefaultServerLocale` is "Default language for system messages and logs." and `DefaultClientLocale` is "Default language for newly created users and pages where the user hasn’t logged in.".

With that in mind `DefaultClientLocale` seams even more fitting given that it's the up front language that every users sees.

Reply:
I still see the first description fits more in this, a message that is send by the system, not necesarilly to an specific user. @jasonblais have you strong opinions here? @jwilander @crspeller what was the original intention of this to config fields?

Reply:
Ok, then I merge it.